openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryHeaderButtons/GlossaryHeaderButtons.component.tsx

```diff
@@ -87,19 +90,10 @@ const GlossaryHeaderButtons = ({
   );
 
   const handleAddGlossaryTermClick = useCallback(() => {
-    if (glossaryFqn) {
-      const activeTerm = glossaryFqn.split(FQN_SEPARATOR_CHAR);
-      const glossary = activeTerm[0];
-      if (activeTerm.length > 1) {
-        history.push(getAddGlossaryTermsPath(glossary, glossaryFqn));
-      } else {
-        history.push(getAddGlossaryTermsPath(glossary));
-      }
-    } else {
-      history.push(
-        getAddGlossaryTermsPath(selectedData.fullyQualifiedName ?? '')
-      );
-    }
+    handleGlossaryTermModalAction(
+      false,
+      !isGlossary ? (selectedData as GlossaryTerm) : undefined
+    );
```

Comment:
Creating two different handler will be a good idea here. Less crowded
`onAddGlossaryTerm`, `onEditGlossaryTerm`
Then you don't need that extra isEdit flag in the params

Reply:
Removed component